### PR TITLE
Implements in-memory loading of images

### LINKED ISSUE #2: Feature request: loading jpeg from memory (ndarray and Tensor)
Probably `jpeg_mem_src` would help (accepts a offset and count).

### PR DESCRIPTION
**Pull request by soumith**
The following now works:

```
with open('chicago.jpg', 'rb') as f:
   img_bytes = f.read()
img = accimage.Image(img_bytes)
```

Fixes #2 cc: @vadimkantorov

### CODE REVIEW

**Comment by vadimkantorov** [comment]
That issue dates a bit back :)

File: accimagemodule.c [hunk 1]
```diff
@@ -241,14 +241,28 @@ static PyTypeObject Image_Type = {
     0,                          /*tp_is_gc*/
 };
 
+#if PY_MAJOR_VERSION == 2
+#define PyUnicode_AsUTF8 PyString_AsString
+#define PyUnicode_Check PyString_Check
```
Comment (willprice) [inline accimagemodule.c]: I'm not sure this is going to work as strings in python 2 were bytes by default: `b'hi' == 'hi'`. This is why travis is failing at the moment. I'm not sure what we should do--drop python 2 support completely? (Although if torchvision is still supporting 2, so should accimage). If we need to support 2, perhaps we could ask for a bytesio object?

Comment (soumith) [reply accimagemodule.c]: I'll fix it, i think I can make the in-memory option be a kwarg-only option probably.

**Comment by soumith** [comment]
i am closing this and calling work bankruptcy :)

**Pull request closed without merging**